doc/sphinx/src/using-closures.rst

```diff
@@ -659,29 +659,40 @@ contains the following member fields, with default values:
 
 .. code-block:: cpp
 
-  struct MixParams {
-    bool verbose = false; // verbosity
-    Real derivative_eps = 3.0e-6;
-    Real pte_rel_tolerance_p = 1.e-6;
-    Real pte_rel_tolerance_e = 1.e-6;
-    Real pte_rel_tolerance_t = 1.e-4;
-    Real pte_abs_tolerance_p = 0.0;
-    Real pte_abs_tolerance_e = 1.e-4;
-    Real pte_abs_tolerance_t = 0.0;
-    Real pte_residual_tolerance = 1.e-8;
-    std::size_t pte_max_iter_per_mat = 128;
-    Real line_search_alpha = 1.e-2;
-    std::size_t line_search_max_iter = 6;
-    Real line_search_fac = 0.5;
-    Real vfrac_safety_fac = 0.95;
-    Real temperature_limit = 1.0e15;
-    Real default_tguess = 300.;
-    Real min_dtde = 1.0e-16;
-    std::size_t pte_small_step_tries = 2;
-    Real pte_small_step_thresh = 1e-16;
-    Real pte_max_dpdv = -1e-8;
-    std::int64_t pte_reduced_system_exclude_idx = -1;  // choose index with largest mass fraction
-  };
+struct MixParams {
+  bool verbose = false;
+  bool iterate_t_guess = true;
+  Real derivative_eps = 3.0e-6;
+  Real pte_rel_tolerance_p = 1.0e-6;
+  Real pte_rel_tolerance_t = 1.0e-4;
+  Real pte_rel_tolerance_e = 1.0e-6;
+  Real pte_rel_tolerance_v = 1.0e-6;
+  Real pte_abs_tolerance_p = 1.0e-6;
+  Real pte_abs_tolerance_t = 1.0e-4;
+
+  Real pte_slow_convergence_thresh = 0.99;
+  Real pte_rel_tolerance_p_sufficient = 1.e2 * pte_rel_tolerance_p;
+  Real pte_rel_tolerance_t_sufficient = 10 * pte_rel_tolerance_t;
+  Real pte_rel_tolerance_e_sufficient = 1.e2 * pte_rel_tolerance_e;
+  Real pte_rel_tolerance_v_sufficient = 1.e2 * pte_rel_tolerance_v;
+  Real pte_abs_tolerance_p_sufficient = 1.e2 * pte_abs_tolerance_p;
+  Real pte_abs_tolerance_t_sufficient = 10 * pte_abs_tolerance_t;
+
+  std::size_t pte_max_iter_per_mat = 128;
+  Real line_search_alpha = 1.e-2;
+  std::size_t line_search_max_iter = 6;
+  Real line_search_fac = 0.5;
+  Real vfrac_safety_fac = 0.95;
+  Real temperature_limit = 1.0e15;
+  Real default_tguess = 300.;
+  Real min_dtde = 1.0e-16;
+  std::size_t pte_small_step_tries = 2;
+  Real pte_small_step_thresh = 1e-16;
+  Real pte_max_dpdv = -1e-8;
+
+  // choose index with largest mass fraction
+  std::int64_t pte_reduced_system_exclude_idx = -1;
+};
```

Comment:
This thing is getting big, is it in global memory?

Reply:
The struct is passed in to a solver when it's constructed, so each solver owns a copy. Are you concerned about that adding up? I think it's still trivial compared to the size of the state vector + scratch.

We could potentially split this into one params object per PTE solver... we could reduce the size that way. Made an issue #550